frontend/public/components/graphs/area.tsx

```diff
@@ -35,13 +39,35 @@ export const AreaChart: React.FC<AreaChartProps> = ({
   query,
   theme = getCustomTheme(ChartThemeColor.blue, ChartThemeVariant.light, areaTheme),
   tickCount = DEFAULT_TICK_COUNT,
+  timespan,
   title,
   xAxis = true,
   yAxis = true,
 }) => {
   const [containerRef, width] = useRefWidth();
+  const [timestamp, setTimestamp] = React.useState(null);
+  React.useEffect(() => {
+    setTimestamp(Date.now());
+  }, [data])
+
   const getLabel = ({x, y}) => `${humanize(y).string} at ${formatDate(x)}`;
-  const container = <ChartVoronoiContainer voronoiDimension="x" labels={getLabel} />;
+  const container = <ChartVoronoiContainer labels={getLabel} voronoiDimension="x" />;
+  const max = data.length ? _.maxBy(data, 'y') : {};
+
+  // Force x-domain to show entire timespan
```

Comment:
I feel like we should use whatever values are in the data whenever we can and only special case when we need. This is only an issue where there's one data point, right?

Reply:
It isn't necessarily only in the 'singe data point' case. If the data domain doesn't span a large enough time period, the x-domain will be smaller than a minute, causing the labels on the x-axis to all be the same (since we don't display seconds).